### PR TITLE
Updated the docs to encourage typed binding usage

### PR DESCRIPTION
**Pull request by bijington**
Documentation update for PR https://github.com/CommunityToolkit/Maui.Markup/pull/156

### CODE REVIEW

File: docs/maui/behaviors/maximum-length-reached-behavior.md [hunk 1]
```diff
@@ -83,7 +83,7 @@ class MaxLengthReachedBehaviorPage : ContentPage
         }.Behaviors(
             new MaxLengthReachedBehavior().Bind(
                 MaxLengthReachedBehavior.CommandProperty,
-                nameof(ViewModel.MaxLengthReachedCommand)));
+                static(ViewModel vm => vm.MaxLengthReachedCommand)));
```
Comment (brminnick) [inline docs/maui/behaviors/maximum-length-reached-behavior.md:86]:
```suggestion
                static (ViewModel vm) => vm.MaxLengthReachedCommand));
```

File: docs/maui/behaviors/user-stopped-typing-behavior.md [hunk 2]
```diff
@@ -85,7 +85,7 @@ class UserStoppedTypingBehaviorPage : ContentPage
         }
         .Bind(
             UserStoppedTypingBehavior.CommandProperty, 
-            nameof(ViewModel. SearchCommand));                    
+            static(ViewModel vm => vm.SearchCommand));
```
Comment (brminnick) [inline docs/maui/behaviors/user-stopped-typing-behavior.md:88]:
```suggestion
            static (ViewModel vm) => vm.SearchCommand);
```

File: docs/maui/converters/color-to-byte-alpha-converter.md [hunk 3]
```diff
@@ -85,18 +85,18 @@ class ColorToByteAlphaConverterPage : ContentPage
...
- 						converter: new ColorToByteAlphaConverter())
- 			}
- 		};
+        {
+            Children =
+            {
+                new Label()
+                    .Text("The alpha component is:"),
+                new Label()
+                    .Bind(
```
Comment (brminnick) [inline docs/maui/converters/color-to-byte-alpha-converter.md:94]: Let's update this binding to use TypedBindings

File: docs/maui/converters/color-to-byte-blue-converter.md [hunk 4]
```diff
@@ -85,18 +85,18 @@ class ColorToByteBlueConverterPage : ContentPage
...
- 						converter: new ColorToByteBlueConverter())
- 			}
- 		};
+        {
+            Children =
+            {
+                new Label()
+                    .Text("The blue component is:"),
+                new Label()
+                    .Bind(
```
Comment (brminnick) [inline docs/maui/converters/color-to-byte-blue-converter.md:94]: Let's update this binding to use TypedBindings

File: docs/maui/converters/color-to-byte-green-converter.md [hunk 5]
```diff
@@ -85,18 +85,18 @@ class ColorToByteGreenConverterPage : ContentPage
...
- 						converter: new ColorToByteGreenConverter())
- 			}
- 		};
+        {
+            Children =
+            {
+                new Label()
+                    .Text("The green component is:"),
+                new Label()
+                    .Bind(
```
Comment (brminnick) [inline docs/maui/converters/color-to-byte-green-converter.md:94]: Let's update this binding to use TypedBindings

File: docs/maui/converters/color-to-byte-red-converter.md [hunk 6]
```diff
@@ -85,18 +85,18 @@ class ColorToByteRedConverterPage : ContentPage
...
- 						converter: new ColorToByteRedConverter())
- 			}
- 		};
+        {
+            Children =
+            {
+                new Label()
+                    .Text("The red component is:"),
+                new Label()
+                    .Bind(
```
Comment (brminnick) [inline docs/maui/converters/color-to-byte-red-converter.md:94]: Let's update this binding to use TypedBindings

File: docs/maui/converters/color-to-cmyk-string-converter.md [hunk 7]
```diff
@@ -85,18 +85,18 @@ class ColorToCmykStringConverterPage : ContentPage
...
-					converter: new ColorToCmykStringConverter())
-		}
-	};
+        {
+            Children =
+            {
+                new Label()
+                    .Text("My favourite Color is:"),
+                new Label()
+                    .Bind(
```
Comment (brminnick) [inline docs/maui/converters/color-to-cmyk-string-converter.md:94]: Let's update this binding to use TypedBindings

File: docs/maui/converters/color-to-cmyka-string-converter.md [hunk 8]
```diff
@@ -85,18 +85,18 @@ class ColorToCmykaStringConverterPage : ContentPage
...
-					converter: new ColorToCmykaStringConverter())
-		}
-	};
+        {
+            Children =
+            {
+                new Label()
+                    .Text("My favourite Color is:"),
+                new Label()
+                    .Bind(
```
Comment (brminnick) [inline docs/maui/converters/color-to-cmyka-string-converter.md:94]: Let's update this binding to use TypedBindings

File: docs/maui/converters/color-to-color-for-text-converter.md [hunk 9]
```diff
@@ -74,11 +74,11 @@ class ColorToColorForTextConverterPage : ContentPage
     public ColorToColorForTextConverterPage()
     {
         Content = new Label { Text = "The Text is showing in an optimum color against the background" }
-			.Bind(
-				Label.TextColorProperty,
-				nameof(ContentPage.BackgroundColor),
-				converter: new ColorToColorForTextConverter(),
-				source: this);
+            .Bind(
```
Comment (brminnick) [inline docs/maui/converters/color-to-color-for-text-converter.md:77]: Let's update this binding to use TypedBindings

File: docs/maui/converters/color-to-degree-hue-converter.md [hunk 10]
```diff
@@ -85,18 +85,18 @@ class ColorToDegreeHueConverterPage : ContentPage
...
- 						converter: new ColorToDegreeHueConverter())
- 			}
- 		};
+        {
+            Children =
+            {
+                new Label()
+                    .Text("The hue component is:"),
+                new Label()
+                    .Bind(
```
Comment (brminnick) [inline docs/maui/converters/color-to-degree-hue-converter.md:94]: Let's update this binding to use TypedBindings

File: docs/maui/converters/color-to-hex-rgb-string-converter.md [hunk 11]
```diff
@@ -85,18 +85,18 @@ class ColorToHexRgbStringConverterPage : ContentPage
...
-					converter: new ColorToHexRgbStringConverter())
-		}
-	};
+        {
+            Children =
+            {
+                new Label()
+                    .Text("My favourite Color is:"),
+                new Label()
+                    .Bind(
```
Comment (brminnick) [inline docs/maui/converters/color-to-hex-rgb-string-converter.md:94]: Let's update this binding to use TypedBindings

File: docs/maui/converters/color-to-hex-rgba-string-converter.md [hunk 12]
```diff
@@ -85,18 +85,18 @@ class ColorToHexRgbaStringConverterPage : ContentPage
...
-					converter: new ColorToHexRgbaStringConverter())
-		}
-	};
+        {
+            Children =
+            {
+                new Label()
+                    .Text("My favourite Color is:"),
+                new Label()
+                    .Bind(
```
Comment (brminnick) [inline docs/maui/converters/color-to-hex-rgba-string-converter.md:94]: Let's update this binding to use TypedBindings

File: docs/maui/converters/color-to-hsl-string-converter.md [hunk 13]
```diff
@@ -85,18 +85,18 @@ class ColorToHslStringConverterPage : ContentPage
...
-					converter: new ColorToHslStringConverter())
-		}
-	};
+        {
+            Children =
+            {
+                new Label()
+                    .Text("My favourite Color is:"),
+                new Label()
+                    .Bind(
```
Comment (brminnick) [inline docs/maui/converters/color-to-hsl-string-converter.md:94]: Let's update this binding to use TypedBindings

File: docs/maui/converters/color-to-hsla-string-converter.md [hunk 14]
```diff
@@ -85,18 +85,18 @@ class ColorToHslaStringConverterPage : ContentPage
...
-					converter: new ColorToHslaStringConverter())
-		}
-	};
+        {
+            Children =
+            {
+                new Label()
+                    .Text("My favourite Color is:"),
+                new Label()
+                    .Bind(
```
Comment (brminnick) [inline docs/maui/converters/color-to-hsla-string-converter.md:94]: Let's update this binding to use TypedBindings

File: docs/maui/converters/color-to-percent-black-key-converter.md [hunk 15]
```diff
@@ -85,18 +85,18 @@ class ColorToPercentBlackKeyConverterPage : ContentPage
...
- 						converter: new ColorToPercentBlackKeyConverter())
- 			}
- 		};
+        {
+            Children =
+            {
+                new Label()
+                    .Text("The key component is:"),
+                new Label()
+                    .Bind(
```
Comment (brminnick) [inline docs/maui/converters/color-to-percent-black-key-converter.md:94]: Let's update this binding to use TypedBindings

File: docs/maui/converters/color-to-percent-cyan-converter.md [hunk 16]
```diff
@@ -85,18 +85,18 @@ class ColorToPercentCyanConverterPage : ContentPage
...
- 						converter: new ColorToPercentCyanConverter())
- 			}
- 		};
+        {
+            Children =
+            {
+                new Label()
+                    .Text("The cyan component is:"),
+                new Label()
+                    .Bind(
```
Comment (brminnick) [inline docs/maui/converters/color-to-percent-cyan-converter.md:94]: Let's update this binding to use TypedBindings

File: docs/maui/converters/color-to-percent-magenta-converter.md [hunk 17]
```diff
@@ -85,18 +85,18 @@ class ColorToPercentMagentaConverterPage : ContentPage
...
- 						converter: new ColorToPercentMagentaConverter())
- 			}
- 		};
+        {
+            Children =
+            {
+                new Label()
+                    .Text("The magenta component is:"),
+                new Label()
+                    .Bind(
```
Comment (brminnick) [inline docs/maui/converters/color-to-percent-magenta-converter.md:94]: Let's update this binding to use TypedBindings

File: docs/maui/converters/color-to-percent-yellow-converter.md [hunk 18]
```diff
@@ -85,18 +85,18 @@ class ColorToPercentYellowConverterPage : ContentPage
...
- 						converter: new ColorToPercentYellowConverter())
- 			}
- 		};
+        {
+            Children =
+            {
+                new Label()
+                    .Text("The yellow component is:"),
+                new Label()
+                    .Bind(
```
Comment (brminnick) [inline docs/maui/converters/color-to-percent-yellow-converter.md:94]: Let's update this binding to use TypedBindings

File: docs/maui/converters/color-to-rgb-string-converter.md [hunk 19]
```diff
@@ -85,18 +85,18 @@ class ColorToRgbStringConverterPage : ContentPage
...
-					converter: new ColorToRgbStringConverter())
-		}
-	};
+        {
+            Children =
+            {
+                new Label()
+                    .Text("My favourite Color is:"),
+                new Label()
+                    .Bind(
```
Comment (brminnick) [inline docs/maui/converters/color-to-rgb-string-converter.md:94]: Let's update this binding to use TypedBindings

File: docs/maui/converters/color-to-rgba-string-converter.md [hunk 20]
```diff
@@ -85,18 +85,18 @@ class ColorToRgbaStringConverterPage : ContentPage
...
-					converter: new ColorToRgbaStringConverter())
-		}
-	};
+        {
+            Children =
+            {
+                new Label()
+                    .Text("My favourite Color is:"),
+                new Label()
+                    .Bind(
```
Comment (brminnick) [inline docs/maui/converters/color-to-rgba-string-converter.md:94]: Let's update this binding to use TypedBindings

File: docs/maui/converters/enum-to-bool-converter.md [hunk 21]
```diff
@@ -140,8 +140,12 @@ class EnumToBoolConverterPage : ContentPage
             Children = 
             {
                 new Picker()
-                    .Bind(Picker.ItemsSourceProperty, nameof(ViewModel.Platforms))
-                    .Bind(Picker.SelectedItemProperty, nameof(ViewModel.SelectedPlatform)),
+                    .Bind(
+                        Picker.ItemsSourceProperty, 
+                        static(ViewModel vm => vm.Platforms))
```
Comment (brminnick) [inline docs/maui/converters/enum-to-bool-converter.md:145]:
```suggestion
                        static (ViewModel vm) => vm.Platforms)
```

File: docs/maui/converters/enum-to-bool-converter.md [hunk 22]
```diff
@@ -140,8 +140,12 @@ class EnumToBoolConverterPage : ContentPage
...
+                    .Bind(
+                        Picker.SelectedItemProperty,
+                        static(ViewModel vm => vm.SelectedPlatform)),
```
Comment (brminnick) [inline docs/maui/converters/enum-to-bool-converter.md:148]:
```suggestion
                        static (ViewModel vm) => vm.SelectedPlatform),
```

File: docs/maui/converters/enum-to-int-converter.md [hunk 23]
```diff
@@ -90,8 +90,12 @@ class EnumToIntConverterPage : ContentPage
     {
         Content = new StackLayout {
           new Picker()
-            .Bind(Picker.ItemSourceProperty, nameof(ViewModel.AllStates)
-            .Bind(Picker.SelectedIndexProperty, nameof(ViewModel.SelectedState),
+            .Bind(
+                Picker.ItemSourceProperty,
+                static(ViewModel vm => vm.AllStates))
```
Comment (brminnick) [inline docs/maui/converters/enum-to-int-converter.md:95]:
```suggestion
                static (ViewModel vm) => vm.AllStates)
```

File: docs/maui/converters/enum-to-int-converter.md [hunk 24]
```diff
@@ -90,8 +90,12 @@ class EnumToIntConverterPage : ContentPage
...
+            .Bind(
+                Picker.SelectedIndexProperty,
+                static(ViewModel vm => vm.SelectedState)),
```
Comment (brminnick) [inline docs/maui/converters/enum-to-int-converter.md:98]:
```suggestion
                static (ViewModel vm) => vm.SelectedState),
```

File: docs/maui/converters/item-tapped-eventargs-converter.md [hunk 25]
```diff
@@ -103,7 +103,9 @@ class ItemTappedEventArgsConverterPage : ContentPage
         {
             HasUnevenRows = true
         }
-        .Bind(ListView.ItemsSourceProperty, nameof(ViewModel.Items))
+        .Bind(
+            ListView.ItemsSourceProperty,
+            static(ViewModel vm => vm.Items))
```
Comment (brminnick) [inline docs/maui/converters/item-tapped-eventargs-converter.md:108]:
```suggestion
            static (ViewModel vm) => vm.Items)
```

File: docs/maui/converters/item-tapped-eventargs-converter.md [hunk 26]
```diff
@@ -112,7 +114,7 @@ class ItemTappedEventArgsConverterPage : ContentPage
             }
             .Bind(
                 EventToCommandBehavior.CommandProperty, 
-                nameof(ViewModel.ItemTappedCommand)));                   
+                static(ViewModel vm => vm.ItemTappedCommand)));
```
Comment (brminnick) [inline docs/maui/converters/item-tapped-eventargs-converter.md:117]:
```suggestion
                static (ViewModel vm) => vm.ItemTappedCommand));   
```

File: docs/maui/converters/selected-item-eventargs-converter.md [hunk 27]
```diff
@@ -101,7 +101,9 @@ class SelectedItemEventArgsConverterPage : ContentPage
         {
             HasUnevenRows = true
         }
-        .Bind(ListView.ItemsSourceProperty, nameof(ViewModel.Items))
+        .Bind(
+            ListView.ItemsSourceProperty,
+            static(ViewModel vm => vm.Items))
```
Comment (brminnick) [inline docs/maui/converters/selected-item-eventargs-converter.md:106]:
```suggestion
            static (ViewModel vm) => vm.Items)
```

File: docs/maui/converters/string-to-list-converter.md [hunk 28]
```diff
@@ -119,23 +119,25 @@ class StringToListConverterPage : ContentPage
...
-		Content = new VerticalStackLayout
+        Content = new VerticalStackLayout
         {
             Children =    
             {
                 new Entry { Placeholder = "Enter some text separated by ',' or '.' or ';'" }
-                    .Bind(Entry.TextProperty, path: nameof(ViewModel.MyValue)),
+                    .Bind(
+                        Entry.TextProperty, 
+                        static(ViewModel vm => vm.MyValue)),
```
Comment (brminnick) [inline docs/maui/converters/string-to-list-converter.md:129]:
```suggestion
                        static (ViewModel vm) => vm.MyValue),
```

File: docs/maui/converters/string-to-list-converter.md [hunk 29]
```diff
@@ -119,23 +119,25 @@ class StringToListConverterPage : ContentPage
...
 
                 new CollectionView
-    			{
-    				ItemTemplate = new DataTemplate(() => new Label().Bind(Label.TextProperty, path: "."))
-    			}.Bind(CollectionView.ItemsSourceProperty,
+                {
+                    ItemTemplate = new DataTemplate(() => new Label().Bind(Label.TextProperty, path: "."))
+                }.Bind(CollectionView.ItemsSourceProperty,
```
Comment (brminnick) [inline docs/maui/converters/string-to-list-converter.md:134]: Let's update this binding to use TypedBindings

File: docs/maui/converters/string-to-list-converter.md [hunk 30]
```diff
@@ -119,23 +119,25 @@ class StringToListConverterPage : ContentPage
...
+                    .Bind(
+                        Entry.TextProperty, 
+                        static(ViewModel vm => vm.MyValue)),
 
                 new CollectionView
-    			{
-    				ItemTemplate = new DataTemplate(() => new Label().Bind(Label.TextProperty, path: "."))
-    			}.Bind(CollectionView.ItemsSourceProperty,
+                {
+                    ItemTemplate = new DataTemplate(() => new Label().Bind(Label.TextProperty, path: "."))
```
Comment (brminnick) [inline docs/maui/converters/string-to-list-converter.md:133]:
```suggestion
                    ItemTemplate = new DataTemplate(() => new Label().Bind(Label.TextProperty, path: Binding.SelfPath))
```

File: docs/maui/layouts/StateContainer.md [hunk 31]
```diff
@@ -70,9 +70,15 @@ Content = new VerticalStackLayout()
     
     new Button()
         .Text("Change State")
-        .Bind(Button.CommandProperty, nameof(StateContainerViewModel.ChangeStateCommand))
-}.Bind(StateContainer.CurrentStateProperty, nameof(StateContainerViewModel.CurrentState))
- .Bind(StateContainer.CanStateChange, nameof(StateContainerViewModel.CanStateChange))
+        .Bind(
+            Button.CommandProperty,
+            static(StateContainerViewModel vm => vm.ChangeStateCommand))
```
Comment (brminnick) [inline docs/maui/layouts/StateContainer.md:75]:
```suggestion
            static (StateContainerViewModel vm) => vm.ChangeStateCommand)
```

File: docs/maui/layouts/StateContainer.md [hunk 32]
```diff
@@ -70,9 +70,15 @@ Content = new VerticalStackLayout()
...
+}.Bind(
+    StateContainer.CurrentStateProperty,
+    static(StateContainerViewModel vm => vm.CurrentState))
```
Comment (brminnick) [inline docs/maui/layouts/StateContainer.md:78]:
```suggestion
    static (StateContainerViewModel vm) => vm.CurrentState)
```

File: docs/maui/layouts/StateContainer.md [hunk 33]
```diff
@@ -70,9 +70,15 @@ Content = new VerticalStackLayout()
...
+ .Bind(
+    StateContainer.CanStateChange,
+    static(StateContainerViewModel vm => vm.CanStateChange))
```
Comment (brminnick) [inline docs/maui/layouts/StateContainer.md:81]:
```suggestion
    static (StateContainerViewModel vm) => vm.CanStateChange)
```

File: docs/maui/markup/extensions/bindable-object-extensions.md [hunk 34]
```diff
@@ -21,22 +21,25 @@ There are a number of overloads for the `Bind` method.
 
 #### Explicit property
 
-A binding from a view model property called `RegistrationCode` to the text property of an `Entry` can be created as follows:
+A binding from a view model (`RegistrationViewModel`) property called `RegistrationCode` to the `Text` property of an `Entry` can be created as follows:
 
 ```csharp
-new Entry().Bind(Entry.TextProperty, nameof(ViewModel.RegistrationCode))
+new Entry().Bind(Entry.TextProperty, static(RegistrationViewModel vm => vm.RegistrationCode))
```
Comment (brminnick) [inline docs/maui/markup/extensions/bindable-object-extensions.md:27]:
```suggestion
new Entry().Bind(Entry.TextProperty, static (RegistrationViewModel vm) => vm.RegistrationCode)
```

File: docs/maui/markup/extensions/bindable-object-extensions.md [hunk 35]
```diff
@@ -46,18 +49,23 @@ The `Bind` method allows for a developer to supply the `Converter` that they wis
 ```csharp
 new Entry()
     .Bind(
-        nameof(ViewModel.RegistrationCode),
+        Entry.TextProperty,
+        nameof(RegistrationViewModel.RegistrationCode),
```
Comment (brminnick) [inline docs/maui/markup/extensions/bindable-object-extensions.md:53]:
```suggestion
        static (RegistrationViewModel vm) => vm.RegistrationCode,
```

File: docs/maui/markup/extensions/bindable-object-extensions.md [hunk 36]
```diff
@@ -46,18 +49,23 @@ The `Bind` method allows for a developer to supply the `Converter` that they wis
...
 See [`TextCaseConverter`](../../converters/text-case-converter.md) for the documentation on it's full usage.
 
 ##### Inline conversion
 
 ```csharp
 new Entry()
     .Bind(
-        nameof(ViewModel.RegistrationCode),
+        Entry.TextProperty,
+        static(RegistrationViewModel vm => vm.RegistrationCode),
```
Comment (brminnick) [inline docs/maui/markup/extensions/bindable-object-extensions.md:68]:
```suggestion
        static (RegistrationViewModel vm) => vm.RegistrationCode,
```

File: docs/maui/markup/extensions/bindable-object-extensions.md [hunk 37]
```diff
@@ -94,8 +102,8 @@ The above could also be written as:
 ```csharp
 new Button()
     .Bind(
-        Button.CommandProperty,
-        nameof(ViewModel.SubmitCommand));
+        Entry.CommandProperty,
+        static(RegistrationViewModel vm => vm.SubmitCommand));
```
Comment (brminnick) [inline docs/maui/markup/extensions/bindable-object-extensions.md:106]:
```suggestion
        static (RegistrationViewModel vm) => vm.SubmitCommand);
```

File: docs/maui/markup/markup.md [hunk 38]
```diff
@@ -35,7 +35,7 @@ entry.SetBinding(Entry.TextProperty, new Binding(nameof(ViewModel.RegistrationCo
 Markup allows us to define the binding fluently and therefore chain multiple methods together to reduce the verbosity of our code:
 
 ```csharp
-new Entry().Bind(Entry.TextProperty, nameof(ViewModel.RegistrationCode))
+new Entry().Bind(Entry.TextProperty, static(ViewModel vm => vm.RegistrationCode))
```
Comment (brminnick) [inline docs/maui/markup/markup.md:38]:
```suggestion
new Entry().Bind(Entry.TextProperty, static (ViewModel vm) => vm.RegistrationCode)
```

File: docs/maui/markup/markup.md [hunk 39]
```diff
@@ -98,7 +98,7 @@ class SampleContentPage : ContentPage
                  .TextColor(Colors.Black)
                  .Height(44)
                  .Margin(5, 5)
-                 .Bind(Entry.TextProperty, nameof(ViewModel.RegistrationCode))
+                 .Bind(Entry.TextProperty, static(ViewModel vm => vm.RegistrationCode))
```
Comment (brminnick) [inline docs/maui/markup/markup.md:101]:
```suggestion
                 .Bind(Entry.TextProperty, static (ViewModel vm) => vm.RegistrationCode)
```

**Review by brminnick — CHANGES_REQUESTED** [review]
Thanks Shaun! Looking good! 

Just FYI - the `Func` syntax was off for all of the typed bindings. My guess is that it's just a bug caused by using Find All + Replace for `nameof`. I fixed the syntax in the ones I saw, but it'd be a good idea for us to double/triple check these again once it's Ready for Review.

File: docs/maui/converters/color-to-byte-alpha-converter.md (hunk 3 above)
Comment (bijington) [reply docs/maui/converters/color-to-byte-alpha-converter.md:94]: We can't as we don't have the converter support yet

File: docs/maui/converters/string-to-list-converter.md (hunk 29 above)
Comment (bijington) [reply docs/maui/converters/string-to-list-converter.md:134]: We can't as we don't have the converter support yet

File: docs/maui/converters/color-to-byte-alpha-converter.md (hunk 3 above)
Comment (brminnick) [reply docs/maui/converters/color-to-byte-alpha-converter.md:94]: Ah, good catch!

File: docs/maui/markup/extensions/bindable-object-extensions.md (hunk 35 above)
Comment (bijington) [reply docs/maui/markup/extensions/bindable-object-extensions.md:67]: We can't as we don't have the converter support yet

**Review by jfversluis — APPROVED** [review]

**Pull request merged**